components/apimgt/org.wso2.carbon.apimgt.api/src/main/java/org/wso2/carbon/apimgt/api/dto/KeyManagerPermissionConfigurationDTO.java

```diff
@@ -0,0 +1,43 @@
+package org.wso2.carbon.apimgt.api.dto;
+
+
+import java.io.Serializable;
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ *KeyManagerPermissionConfiguration model
+ */
+public class KeyManagerPermissionConfigurationDTO implements Serializable {
+
+    private String permissionType = null;
+    private List<String> roles = new ArrayList<String>();
+
+    public KeyManagerPermissionConfigurationDTO () {
+        this.setPermissionType("PUBLIC");
+    }
+
+    public KeyManagerPermissionConfigurationDTO(String permissionType, List<String> roles) {
+        this.permissionType = permissionType;
+        this.roles = roles;
+    }
+
+    public String getPermissionType () {
+        return permissionType;
+    }
+
+    public void setPermissionType (String permissionType) {
+        this.permissionType = permissionType;
+    }
+
+    public List<String> getRoles() {
+        return roles;
+    }
+
+    public void setRoles(List<String> roles) {
+        if (roles == null) {
+            roles = new ArrayList<String>();
```

Comment:
We can return if the roles are null right? Since the roles instance variable is a String array list already.

Reply:
Fixed in cfc856942d5dcfb5dcdcc9eb904b5de6a6cdb495